Assets/Scripts/Augment/AugmentImplementations/DDRBody.cs

```diff
@@ -0,0 +1,185 @@
+
+    [SerializeField]
+    private AugmentAnimator animator;
+
+    private Material ddrMaterial;
+
+    private const int screenMaterialIndex = 2;
+
+    private float arrowHeigth = 0;
+
+    private int arrowDirectionQuadrant = 0;
+
+    private const float screenHeigth = 3.5f;
+    private const float errorMarginInput = 0.1f;
+
+    private const float targetPoint = 3;
+
+    [SerializeField]
+    private Precision[] precisionsGoodToBad;
+
+    LTDescr arrowMover;
+
+    [SerializeField]
+    private float secondPerArrow = 1;
+    private float musicPaces = 0.4285714f;
+
+
+    public override void Start()
+    {
+        meshRenderer.materials[screenMaterialIndex] = Instantiate(meshRenderer.materials[screenMaterialIndex]);
+        ddrMaterial = meshRenderer.materials[screenMaterialIndex];
+        precisionText.enabled = false;
+
+        gunController = transform.parent.GetComponent<GunController>();
+        if (!gunController)
+        {
+            Debug.Log("DanceBody not attached to gun parent!");
+            return;
+        }
+
+        if (gunController.player)
+        {
+            gunController.player.inputManager.onFirePerformed += Fire;
+            gunController.player.inputManager.onMovePerformed += ArrowSelect;
+            arrowMover = LeanTween.value(gameObject, SetArrowHeigth, 0, screenHeigth, secondPerArrow)
+                .setRepeat(-1)
+                .setOnComplete(ResetArrow);
+
+            LeanTween.value(gameObject, SetBackgroundZoom, 0.5f, 1.5f, musicPaces)
+                .setLoopPingPong()
+                .setOnComplete(
+                () => animator.OnFire(0));
+
+            animator.OnInitialize(gunController.stats);
+            
+        }
+
+    }
+
+    protected override void Reload(GunStats stats)
+    {
+        Precision? precision = null;
+
+        if (Mathf.Abs(arrowHeigth - targetPoint) <= precisionsGoodToBad[0].range)
+            precision = precisionsGoodToBad[0];
+        else if (Mathf.Abs(arrowHeigth - targetPoint) <= precisionsGoodToBad[1].range)
+            precision = precisionsGoodToBad[1];
+        else if (Mathf.Abs(arrowHeigth - targetPoint) <= precisionsGoodToBad[2].range)
+            precision = precisionsGoodToBad[2];
+        else if (Mathf.Abs(arrowHeigth - targetPoint) <= precisionsGoodToBad[3].range)
+            precision = precisionsGoodToBad[3];
+
+        if (!precision.HasValue)
+            return;
+
+        precisionText.enabled = true;
+        precisionText.text = precision.Value.text;
+        precisionText.color = precision.Value.color;
+        precisionText.gameObject.LeanScale(new Vector3(1.1f, 1.1f, 1.1f), 0.5f)
+            .setEasePunch()
+            .setOnComplete(
+            () => precisionText.enabled = false);
+            
+        LeanTween.value(gameObject, SetFlashFactor, 0, 20f, 0.5f).setEasePunch();
+
+        gunController.Reload(reloadEfficiencyPercentage*precision.Value.awardFactor);
+        animator.OnReload(1);
```

Comment:
So um... Why do you call `OnReload` with a bogus argument? Perhaps this only happens because there_is_ an argument in the `OnReload` method in `BarrelAnimator`. Does it make that much sense to use an integer for this on a gun body, when it doesn't even care about the magazine size?

Reply:
The problem with this is two-fold:
- There might (likely) exist barrels thad doesn't care about magazineSize, but wants to do something OnReload
- I believe BarrelAnimator can and should be generalized to an augmentAnimator with optional parameters instead

I didn't want to make such a change without talking it through, and especially not in this PR due to it's already large size